### ISSUE TITLE
Indexes are dropped when changing the table configuration through `rethinkdb.table_config`

### ISSUE THREAD

**Issue by VeXocide** [rethinkdb/rethinkdb#4533]
When changing the table configuration through `rethinkdb.table_config` the indexes get dropped as they're not copied correctly.

```
r.tableCreate("x")
r.table("x").indexCreate("x")
r.table("x").indexList() // returns ["x"]
r.db("rethinkdb").table("table_config").filter({"db": "test", "name": "x"}).update({"write_acks": "single"})
r.table("x").indexList() // returns []
```

**Comment by coffeemug** [comment]
I think we should definitely fix this before shipping the beta. This is a pretty big deal.

**Comment by VeXocide** [comment]
I've just checked the other properties in the `table_config_t` and they're fine thus I'll be fixing this as part of #4525.

**Comment by danielmewes** [comment]
Note: We also need to make sure this works with emergency repair.

**Comment by VeXocide** [comment]
Emergency repair is fine, see https://github.com/rethinkdb/rethinkdb/blob/next/src/clustering/table_contract/emergency_repair.cc#L114

**Comment by VeXocide** [comment]
In CR 3074 by @danielmewes.

**Comment by coffeemug** [comment]
:clap:

**Comment by VeXocide** [comment]
Merged into `next` via commit  1f224936ba57e8b6292b5326c7f1c6aaa7c67899.
